plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java

```diff
@@ -102,13 +129,53 @@ public DockerConnectorConfiguration(URI dockerDaemonUri,
         this.networkFinder = networkFinder;
     }
 
-    public static String getExpectedLocalHost() {
-        return SystemInfo.isLinux() ? "localhost" : dockerDaemonUri().getHost();
+    public static boolean isUnixSocketUri(URI uri) {
```

Comment:
It does the half part of the contract, as it doesn't say about method reaction on null parameter.
Should i expect `NullPointerException`, `IllegalArgumentException` or maybe `BadRequestException`, from the `@NotNull` annotation it is not clear + i don't think that javadoc generators respect this annotation.